Add a sort button for playlist

uiryuu opened the pull request:
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #5162, #4095.

---

**Description:**
Added a sort button, when clicking on it, a context menu will prompt the user to choose the sorting method. We could add more sorting methods in the future. For now, I only added sorting ascending and descending by file path.

The way I implemented the playlist reordering is a bit strange, that's because the only operations I have to manipulate the playlist are removing/adding item to the playlist, and moving an item to a specific index. If I use moving index, then the best time complexity I can achieve is O(n^2). So I remove all the playlist items (except the one is current playing) and add them back by the order we want.

This PR also refactors private playlist-related functions from player core to `MPVCommandWrapper`, which is an extension of `MPVController`. We should avoid directly call methods in `MPVController` outside of player core in the future.

uiryuu committed:
playlist: add sort button & menu, fix #5162 #4095

uiryuu committed:
playlist: add fallback image for sort button for macOS 10.15

low-batt commented on iina/PlaylistViewController.swift:
```diff
@@ -468,6 +474,28 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource, NSTableVi
+  @IBAction func sortingBtnAction(_ sender: NSButton) {
+    let menu = NSMenu()
+    if #available(macOS 14.0, *) {
+      menu.addItem(.sectionHeader(title: "Sorting"))
```
Needs localization.

uiryuu replied:
Added.

low-batt commented on iina/PlaylistViewController.swift:
```diff
@@ -468,6 +474,28 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource, NSTableVi
+    if #available(macOS 14.0, *) {
+      menu.addItem(.sectionHeader(title: "Sorting"))
+    }
+    menu.addItem(withTitle: NSLocalizedString("playlist.sorting.path_ascending", comment: "File Path (A-Z)"), action: #selector(sortPathAscending), keyEquivalent: "")
```
Files don't have to start with a letter, so A-Z isn't really appropriate. Maybe "Ascending/Descending"? I do like that (A-Z) is nice and short.

uiryuu replied:
Changed to "Ascending/Descending", see whether or not it looks better to you.

low-batt reviewed (changes requested):
Tuesday is my busy day, so just some quick possibly incoherent thoughts…

PR #5173, already on the 1.4.0-beta2 schedule, does something similar. This PR contains some improvements over that one.

The new button is missing a help tag, which the other two buttons have:
https://github.com/iina/iina/blob/7b4a7bda0a6c2a99c4ba98f874a1365f25a360cd/iina/PlaylistViewController.swift#L105-L106

Sorting based on file path is a bit confusing when titles are being displayed.

One concern I have about sorting is that apparently IPTV playlists can be very large. Will IINA lock up if someone tries to sort a large playlist? See issue #3357. I don't think this is a blocking factor because this isn't just a button you can accidentally hit as it pops up a menu. That makes it less of a concern.

I guess my biggest concern is that sorting by file path while displaying titles makes the feature seem like it is broken.

uiryuu committed:
Merge branch 'develop' into sort-playlist

uiryuu committed:
playlist: add sorting for displayname

Updated based on review. Added localizations for several strings, added tooltip for the new button.

Co-Authored-By: Karan Sah <58397645+karan0046@users.noreply.github.com>

uiryuu commented:
> PR https://github.com/iina/iina/pull/5173, already on the 1.4.0-beta2 schedule, does something similar.

I've added @karan0046 as coauthor in the followup commit to give them credit after this PR is merged.

> The new button is missing a help tag, which the other two buttons have:

Added.

> Sorting based on file path is a bit confusing when titles are being displayed.

I was afraid that the `filenameForDisplay` property will change depending on whether or not the user choose to show song title from metadata in the playlist. But `filenameForDisplay` is just the last path component for a file, so I was wrong. Added sort by filename in the latest commit.

> I guess my biggest concern is that sorting by file path while displaying titles makes the feature seem like it is broken.

Still want to keep sort by file path as maybe some users would like to sort based on the folder structure.
